Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -352,69 +505,102 @@ def get_random_redirection(original: Transition, all_redirections: Iterable[Tran
+
+        index = len(closed_door_levels)  # we skip the closed_door_levels, as they are already done
         while index < len(level_list):
             choice = random.choice(tuple(Choice))
+
+            # Option 1: connect to one or more existing levels
             if total_con_left > 0 and (
-                __connections_left[level_list[index].area_id] == 1 or choice == Choice.CONNECT
+                len(loose_ends) > 0
+                or __connections_left[level_list[index].area_id] == 1
+                or choice == Choice.CONNECT
             ):
-                # option 1: connect to one or more existing levels
-                connect_to_existing(level_list, index, link_list)
-            elif __connections_left[level_list[index].area_id] > 1 and (
-                total_con_left == 0 or choice == Choice.INBETWEEN
-            ):
-                # option 2: put the current level inbetween an already established connection
                 total_con_left += __connections_left[level_list[index].area_id]
-                level_a, level_b = link_list.pop(random.randrange(len(link_list)))
-                unlink_two_levels(level_a, level_b)
-                link_list.extend((
-                    link_two_levels(level_list[index], level_a),
-                    link_two_levels(level_list[index], level_b),
-                ))
-                total_con_left -= 2
+                connect_to_existing(index, level_list)
+
+            # Option 2: put the current level inbetween an already established connection
+            elif __connections_left[level_list[index].area_id] > 1:
+                total_con_left += __connections_left[level_list[index].area_id]
+                link_chosen = random.choice(link_list)
+                insert_area_inbetween(link_chosen[0], link_chosen[1], level_list[index].area_id)
+
+            # Option 3: break open a connection that's part of a loop, then restart iteration
             else:
-                # option 3: break open a connection that's part of a level loop,
-                # then restart this iteration
-                break_open_connection(level_list, index, link_list)
+                try:
+                    find_and_break_open_connection(link_list)
+                except NoConnectionFoundError:
+                    print(" ")
+                    print("ATTENTION ATTENTION: THE RANDOMIZER CRASHED!!!")
+                    print("Please notify a developer!")
+                    print(" ")
+                    break
                 continue
+
             index += 1
 
-        link_list_to_transitions(
-            link_list,
-            transitions_map,
-            _possible_origins_bucket,
-            _possible_redirections_bucket,
-        )
+        # Once the link_list is completed, it's time to fill the transitions_map:
+        for link in link_list:
+            transitions_map[link[0]] = link[1]
```

Comment:
A MutableMapping can be updated with an iterable of two-pairs (like a list or 2-tuples) OR anything that has key-values (like a dict)
```suggestion
        transitions_map.update(link_list)
```

Reply:
I most certainly didn't know this one.
Personally I would prefer to keep it the old way because your proposed method doesn't feel intuitive to me at all, but hey you make the rules here I guess.
Changed.